client/src/use/useSettings.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  reactive, ref,
+} from '@vue/composition-api';
+
+export interface NewTrackSettings {
+    mode: string;
+    type: string;
+    modeSettings: {
+      Track: {
+        autoAdvanceFrame: boolean;
+      };
+      Detection: {
+        continuous: boolean;
+      };
+    };
+  }
+export interface ClientSettings {
+  NewTrackSettings: NewTrackSettings;
+}
+
+export default function useSettings() {
+  const clientSettings = reactive({
+    newTrackSettings: {
+      mode: ref('Track'),
+      type: ref('unknown'),
+      modeSettings: {
+        Track: {
+          autoAdvanceFrame: ref(false),
+        },
+        Detection: {
+          continuous: ref(true),
+        },
+      },
+    },
+  });
```

Comment:
```suggestion
  }) as ClientSettings;
```

After you add that, you'll have to fix the type errors from `NewTrackSettings` not being pascalCase.